nixos/modules/services/matrix/synapse.nix

```diff
@@ -346,8 +490,8 @@ in {
 
             log_config = mkOption {
               type = types.path;
-              default = ./synapse-log_config.yaml;
-              defaultText = lib.literalExpression "nixos/modules/services/matrix/synapse-log_config.yaml";
+              default = genLogConfigFile "synapse";
```

Comment:
That's a good idea! We'd have to special-case this option during config file generation though to allow this as currently the config file is basically just `(pkgs.format.yaml {}).generate "homeserver.yaml" config.services.matrix-synapse.settings`. Do you think this would be worth it?

Reply:
Hah, totally misread that, sorry :sweat_smile: 

ftr my point still stands, I think the current changes to log_config are still an improvement.

My preferred solution would be something like `services.matrix-synapse.logConfig.defaults` (and then perhaps the possibility to define custom settings per worker-name, i.e. `services.matrix-synapse.logConfig.federation-sender`, just a first idea though). I think that's preferable over special-casing the format renderer because this seems like too much of a deviation from how this rfc42 stuff works in other modules which might be rather unexpected.

However, I'm resolving this now. As I said, I consider this state an improvement over the previous stuff already and more sophisticated changes to logging in a PR about workers would be a clear case of feature creep IMHO. Feel free to file a follow-up PR with that if you want to, I'll add it to my todolist, perhaps I'll get to it soonish otherwise :)